overlord/hookstate/hookmgr.go

```diff
@@ -295,6 +318,19 @@ func runHookAndWait(snapName string, revision snap.Revision, hookName, hookConte
 		return nil, err
 	}
 
+	// add timeout handling
+	var killTimer *time.Timer
+	if maxRuntime > 0 {
+		killTimer = time.AfterFunc(maxRuntime, func() {
+			if err := killemAll(command); err != nil {
+				logger.Noticef("cannot kill %q: %s", hookName, err)
+			}
+			// ensure we log what happened
+			fmt.Fprintf(buffer, "\nexceeded maximum runtime of %s\n", maxRuntime)
```

Comment:
I think this is problematic, because I don' think buffers are goroutine safe,  setting them on command works because there are dedicated goroutines setup but here is yet a different one. buffer can be used again only after command.Wait() I think... so if we really want to add this safely we need to do a bit more work